### PR TITLE
[CLI Config Parity] migrate sequential read size mb flag and test GCS Connection configs

### PR DESCRIPTION
**Pull request by ashmeenkaur**
### Description
migrate sequential read size mb flag and test GCS Connection configs.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - Updated
3. Integration tests - NA

### CODE REVIEW

File: cfg/constants.go [hunk 1]
```diff
@@ -33,3 +33,8 @@ const (
 	// ExperimentalMetadataPrefetchOnMountAsynchronous is the prefetch-mode where mounting is marked complete once prefetch has started.
 	ExperimentalMetadataPrefetchOnMountAsynchronous = "async"
 )
+
+const (
+	// MaxSequentialReadSizeMb is the max value supported by sequential-read-size-mb flag.
+	MaxSequentialReadSizeMB = 1024
```
Comment (kislaykishore) [inline cfg/constants.go:39]: Does it really have to be exported? All the usages seem to be confined to the same package.

File: cfg/validate.go [hunk 2]
```diff
@@ -72,6 +72,13 @@ func IsValidExperimentalMetadataPrefetchOnMount(mode string) error {
 	}
 }
 
+func isValidSequentialReadSizeMB(sequentialReadSizeMB int64) error {
+	if sequentialReadSizeMB < 1 || sequentialReadSizeMB > MaxSequentialReadSizeMB {
+		return fmt.Errorf("SequentialReadSizeMb should be between 1 than %d", MaxSequentialReadSizeMB)
```
Comment (kislaykishore) [inline cfg/validate.go:77]:
```suggestion
		return fmt.Errorf("SequentialReadSizeMb should be between 1 and %d", MaxSequentialReadSizeMB)
```

File: cfg/validate.go [hunk 3]
```diff
@@ -96,5 +103,9 @@ func ValidateConfig(config *Config) error {
 		return fmt.Errorf("error parsing token-url config: %w", err)
 	}
 
+	if err = isValidSequentialReadSizeMB(config.GcsConnection.SequentialReadSizeMb); err != nil {
+		return fmt.Errorf("error parsing token-url config: %w", err)
```
Comment (kislaykishore) [inline cfg/validate.go:107]: typo
```suggestion
		return fmt.Errorf("error parsing gcs-connection config: %w", err)
```

File: cfg/validate_test.go [hunk 4]
```diff
@@ -145,6 +189,38 @@ func TestValidateConfigUnsuccessful(t *testing.T) {
...
+				GcsConnection: GcsConnectionConfig{
+					SequentialReadSizeMb: 2048,
+				},
+				MetadataCache: MetadataCacheConfig{
+					ExperimentalMetadataPrefetchOnMount: "sync",
+				},
+			},
+		},
+		{
+			name: "Sequential read size MB more than 1024 (max permissible value)",
```
Comment (kislaykishore) [inline cfg/validate_test.go:214]: typo
```suggestion
			name: "Sequential read size MB less than 1 (min permissible value)",
```

File: cfg/validate.go [hunk 5]
```diff
@@ -72,6 +72,13 @@ func IsValidExperimentalMetadataPrefetchOnMount(mode string) error {
 	}
 }
 
+func isValidSequentialReadSizeMB(sequentialReadSizeMB int64) error {
```
Comment (kislaykishore) [inline cfg/validate.go:75]: nit: Avoid repetition - https://google.github.io/styleguide/go/decisions#repetition
```suggestion
func isValidSequentialReadSizeMB(size int64) error {
```

File: cfg/validate.go [hunk 6]
```diff
@@ -72,6 +72,13 @@ func IsValidExperimentalMetadataPrefetchOnMount(mode string) error {
 	}
 }
 
+func isValidSequentialReadSizeMB(size int64) error {
+	if sequentialReadSizeMB < 1 || sequentialReadSizeMB > maxSequentialReadSizeMB {
```
Comment (ashmeenkaur) [inline cfg/validate.go:76]:
```suggestion
	if size < 1 || size > maxSequentialReadSizeMB {
```

File: cmd/root_test.go [hunk 7]
```diff
@@ -455,3 +456,108 @@ func TestArgsParsing_GCSAuthFlagsThrowsError(t *testing.T) {
...
+				assert.Equal(t, tc.expectedConfig.GcsConnection, gotConfig.GcsConnection)
+			}
+		})
+	}
+}
+func TestArgsParsing_GCSConnectionFlagsThrowsError(t *testing.T) {
+	tests := []struct {
+		name           string
+		args           []string
+		expectedConfig *cfg.Config
```
Comment (kislaykishore) [inline cmd/root_test.go:528]:
```suggestion
```

File: cmd/root_test.go [hunk 8]
```diff
@@ -455,3 +456,108 @@ func TestArgsParsing_GCSAuthFlagsThrowsError(t *testing.T) {
...
+					SequentialReadSizeMb:       200,
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var gotConfig *cfg.Config
+			cmd, err := NewRootCmd(func(cfg *cfg.Config, _ string, _ string) error {
```
Comment (kislaykishore) [inline cmd/root_test.go:509]: Nit:
```suggestion
			cmd, err := NewRootCmd(func(cfg *cfg.Config, _, _ string) error {
```

File: cmd/root_test.go [hunk 9]
```diff
@@ -455,3 +456,108 @@ func TestArgsParsing_GCSAuthFlagsThrowsError(t *testing.T) {
...
+		},
+		{
+			name: "Invalid value for http-client-timeout flag",
+			args: []string{"gcsfuse", "--http-client-timeout=200", "abc", "pqr"},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			cmd, err := NewRootCmd(func(cfg *cfg.Config, _ string, _ string) error {
```
Comment (kislaykishore) [inline cmd/root_test.go:554]: Nit:
```suggestion
			cmd, err := NewRootCmd(func(cfg *cfg.Config, _, _ string) error {
```

File: cmd/root_test.go (hunk 8 above)
Comment (ashmeenkaur) [reply cmd/root_test.go:509]: Oops, sorry.. forgot to do this on this PR after rebasing.

**Review by kislaykishore — APPROVED** [review]

**Pull request merged**